ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -33,23 +34,43 @@ def __init__(self):
 
         rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)
         rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)
+        rospy.Subscriber('/traffic_waypoint', Int32, self.traffic_cb)
+        rospy.Subscriber('/obstacle_waypoint', Lane, self.obstacle_cb)
 
-        # TODO: Add a subscriber for /traffic_waypoint and /obstacle_waypoint
-
-        self.final_waypoints_pub = rospy.Publisher('final_waypoints', Lane,
-                                                   queue_size=1)
+        self.pub = rospy.Publisher('final_waypoints', Lane, queue_size=1)
 
         # TODO: Add other member variables you need below
+        self.world = None
+        self.ego = None
 
+        self.publish()
         rospy.spin()
 
+    def publish(self):
+        rate = rospy.Rate(5)
+        while not rospy.is_shutdown():
+            idx = self.find_next_waypoint()
+            if idx > -1:
+                rospy.loginfo("Next waypoint: {}".format(idx))
+                waypoints = self.world.waypoints + self.world.waypoints
+                waypoints = waypoints[idx:idx+LOOKAHEAD_WPS]
+                for i in range(LOOKAHEAD_WPS):
+                    self.set_waypoint_velocity(waypoints, i, 20)
+                lane = Lane()
+                lane.header.frame_id = '/world'
+                lane.header.stamp = rospy.Time.now()
+                lane.waypoints = waypoints
+                self.pub.publish(lane)
+            rate.sleep()
+
     def pose_cb(self, msg):
-        # TODO: Implement
-        pass
+        if self.ego is None or self.ego.header.seq < msg.header.seq:
+            # TODO: Calculate ego's velocity (and maybe acceleration?) if needed
+            self.ego = msg
 
     def waypoints_cb(self, waypoints):
-        # TODO: Implement
-        pass
+        if self.world is None or self.world.header.seq < waypoints.header.seq:
+            self.world = waypoints
```

Comment:
I believe waypoints does not change so can keep first one instead of updating every time.

Reply:
Is that a reasonable assumption? I mean, the current publisher to that topic doesn't seem to change the waypoints, however relying on that will make us highly coupled to the behaviour of the publisher.

Reply:
This is from project description: 
/base_waypoints | styx_msgs/Lane | Waypoints as provided by a static .csv file.